src/DynamoPackages/PackageManagerExtension.cs

```diff
@@ -149,6 +159,25 @@ private void OnMessageLogged(ILogMessage msg)
                 this.MessageLogged(msg);
             }
         }
+        
+        private void OnCurrentWorkspaceChanged(IWorkspaceModel ws)
+        {
+            if (ws is WorkspaceModel)
+            {
+                if (currentWorkspace != null)
+                {
+                    (currentWorkspace as WorkspaceModel).CollectingPackageDependencies -= GetLocalPackages;
```

Comment:
hmm, are you sure you want to do this when a user opens a custom node workspace for example? Is what you really want to unsubscribe when the user closes the home work space? - I am 100% not sure - just curious if you considered custom node workspaces.

Reply:
Don't we want `PackageDependencies` serialization to work for both .dyn and .dyf? My intention here was that if you save while in a custom node workspace (i.e. while `currentWorkspace` is a custom node), it should save the `PackageDependencies` of the custom node.

Reply:
Alternatively, we could put all `PackageDependencies`, including those of custom nodes the graph depends on, up in the .dyn.

Reply:
That seems weird to me though

Reply:
I think your intention sounds better than the second description since we discussed top level dependencies only.